src/account.ts

```diff
@@ -534,90 +536,51 @@ export class Account {
         if (ready) {
+
             const feeEst = await this.getZpClient().feeEstimate([amount], TxType.Deposit, false);
             const relayerFee = feeEst.relayerFee;
             console.info(`Using relayer fee: base = ${relayerFee.fee}, perByte = ${relayerFee.oneByteFee}`);
-
-            let totalApproveAmount = await this.getZpClient().shieldedAmountToWei(amount + feeEst.total);
-            const currentAllowance = await this.getClient().allowance(this.getTokenAddr(), this.getPoolAddr());
-            if (totalApproveAmount > currentAllowance) {
-                totalApproveAmount -= currentAllowance;
-                console.log(`Increasing allowance for the Pool (${this.getPoolAddr()}) to spend our tokens (+ ${await this.weiToHuman(totalApproveAmount)} ${this.tokenSymbol()})`);
-                await this.getClient().increaseAllowance(this.getTokenAddr(), this.getPoolAddr(), totalApproveAmount.toString());
-            } else {
-                console.log(`Current allowance (${await this.weiToHuman(currentAllowance)} ${this.tokenSymbol()}) is greater or equal than needed (${await this.weiToHuman(totalApproveAmount)} ${this.tokenSymbol()}). Skipping approve`);
+                        
+            const depositScheme = this.config.pools[this.getCurrentPool()].depositScheme;
+            let totalNeededAmount = await this.getZpClient().shieldedAmountToWei(amount + feeEst.total);
+            if (depositScheme == DepositType.Approve) {
+                // check a token approvement if needed (in case of approve deposit scheme)
+                const currentAllowance = await this.getClient().allowance(this.getTokenAddr(), this.getPoolAddr());
+                if (totalNeededAmount > currentAllowance) {
+                    totalNeededAmount -= currentAllowance;
+                    console.log(`Increasing allowance for the Pool (${this.getPoolAddr()}) to spend our tokens (+ ${await this.weiToHuman(totalNeededAmount)} ${this.tokenSymbol()})`);
+                    await this.getClient().increaseAllowance(this.getTokenAddr(), this.getPoolAddr(), totalNeededAmount.toString());
+                } else {
+                    console.log(`Current allowance (${await this.weiToHuman(currentAllowance)} ${this.tokenSymbol()}) is greater or equal than needed (${await this.weiToHuman(totalNeededAmount)} ${this.tokenSymbol()}). Skipping approve`);
+                }
+            } else if (depositScheme == DepositType.PermitV2) {
+                const currentAllowance = await this.getClient().allowance(this.getTokenAddr(), PERMIT2_CONTRACT);
+                if (totalNeededAmount > currentAllowance) {
+                    const maxTokensAmount = 2n ** 256n - 1n;
+                    console.log(`Approving Permit2 contract (${PERMIT2_CONTRACT}) to spend max amount of our tokens`);
+                    await this.getClient().approve(this.getTokenAddr(), PERMIT2_CONTRACT, maxTokensAmount.toString());
```

Comment:
isn't it better to make unlimited allowance? 
https://docs.uniswap.org/contracts/permit2/overview#approving-permit2